irods/query.py

```diff
@@ -65,7 +67,16 @@ def add_keyword(self, keyword, value = ''):
 
     def filter(self, *criteria):
         new_q = self._clone()
-        new_q.criteria += list(criteria)
+
+        if self.case_sensitive:
+            new_q.criteria += list(criteria)
+        else:
+            for criterion in criteria:
+                if type(criterion.unprocessed_value) == str:
```

Comment:
Is this because there times when the type will not be a `str` ?

Reply:
The criterion class doesn't appear to do any type checking of the value, so I think we need to take into account that values may have unexpected types.